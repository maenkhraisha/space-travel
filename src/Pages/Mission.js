import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMission } from '../redux/mission/mission';
import './mission.css';
import MissionList from '../Components/missionList';

const url = 'https://api.spacexdata.com/v3/missions';
const Mission = () => {
  const dispatch = useDispatch();
  const { missions: missionArray } = useSelector((state) => state.mission);

  useEffect(() => {
    fetchMission(url, dispatch);
  }, []);

  return (
    <div>
      <table className="body">
        <thead className="tablehead">
          <tr>
            <th className="tableRow">Misson</th>
            <th className="tableRow">Description</th>
            <th className="tableRow">Status</th>
            <th className="tableRow">{}</th>
          </tr>
        </thead>
        <tbody>
          {missionArray.map((mission) => (
            <MissionList key={mission.mission_id} mission={mission} />
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Mission;
