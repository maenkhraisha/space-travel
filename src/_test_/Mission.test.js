import React from 'react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Mission from '../Pages/Mission'
import store from '../redux/configureStore';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Mission />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});