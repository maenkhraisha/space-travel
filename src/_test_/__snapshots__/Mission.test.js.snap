// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <table
    className="body"
  >
    <thead
      className="tablehead"
    >
      <tr>
        <th
          className="tableRow"
        >
          Misson
        </th>
        <th
          className="tableRow"
        >
          Description
        </th>
        <th
          className="tableRow"
        >
          Status
        </th>
        <th
          className="tableRow"
        />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
